datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -749,7 +807,7 @@ def event(self, event):
         for key in event:
             # Ensure strings have the correct type
             try:
-                event[key] = to_string(event[key])
+                event[key] = to_string(event[key])  # type: ignore
```

Comment:
Not related to this PR, but this is weird, we are passing list of tags to `to_string()`, hopefully it deals with that fine.

Reply:
Well, it *does* deal with it fine:

```python
>>> from datadog_checks.base import to_string
>>> to_string(['hello', 'world'])
['hello', 'world']
```

But that's only because `isinstance(['hello', 'world'], bytes)` evaluates to `False`. 🤣🙀

There used to be an explicit `isinstance(value, bytes)`: https://github.com/DataDog/integrations-core/pull/3822/files#diff-874d5159f166811f6c78ff097251312cR544-R554

but it was lost when we merged the two `AgentCheck` classes together: https://github.com/DataDog/integrations-core/pull/5379/files#diff-874d5159f166811f6c78ff097251312cR716-R722

Will bring it back here.

Reply:
Filed https://github.com/DataDog/integrations-core/pull/5996

Reply:
Check type of values before coercing to text: https://github.com/DataDog/integrations-core/pull/5997